src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -101,19 +102,23 @@ export const Deposit: React.FC<DepositProps> = ({
 
   const getDepositGasEstimate = useCallback(
     async (deposit: DepositValues): Promise<string | undefined> => {
-      if (!(userAddress && assetReference && yearnInvestor && accountId && opportunityData)) return
+      if (!(userAddress && assetReference && accountId && opportunityData)) return
       try {
-        const yearnOpportunity = await yearnInvestor.findByOpportunityId(opportunityData.assetId)
-        if (!yearnOpportunity) throw new Error('No opportunity')
-        const preparedTx = await yearnOpportunity.prepareDeposit({
-          amount: bnOrZero(deposit.cryptoAmount).times(`1e+${asset.precision}`).integerValue(),
-          address: fromAccountId(accountId).account,
-        })
-        // TODO(theobold): Figure out a better way for the safety factor
-        return bnOrZero(preparedTx.gasPrice)
-          .times(preparedTx.estimatedGas)
-          .integerValue()
-          .toString()
+        const amountCryptoBaseUnit = bnOrZero(deposit.cryptoAmount).times(
+          bn(10).pow(asset.precision),
+        )
+        const quote = await getThorchainSaversQuote(asset, amountCryptoBaseUnit)
+        const chainAdapters = getChainAdapterManager()
+        const adapter = chainAdapters.get(chainId) as unknown as UtxoBaseAdapter<UtxoChainId>
```

Comment:
you can deposit eth?

Reply:
this might be technically wrong, but still work with a comment?

Reply:
787ebf41ad14468cf6f5b592b807fbe034dc298a